doc/template_definition.md

```diff
@@ -81,8 +124,79 @@ method.getBody().insertBegin(injectedCode);
 
 ```
 
-Kinds of templating
--------------------
+Definition of templates using `PatternBuilder`
+--------------------------------------------------
+
+The body of method `CheckBoundTemplate#statement` below defines a Spoon template. 
+
+```java
+public class CheckBoundTemplate /*it doesn't matter what it extends or implements*/ {
+	// it doesn't matter what other type members are in template class
+	// it doesn't matter what is the name of the method whose body will be used as template
+	public void statement(Collection<?> _col_) {
+		if (_col_.size() > 10)
+			throw new OutOfBoundException();
+	}
+}
+```
+
+The code, which creates a Spoon template using `PatternBuilder` looks like this:
+```java
+Pattern t = PatternBuilder.create(factory,
+	//defines template class.
+	CheckBoundTemplate.class, 
+	//defines which part of template class will be used as template model
+	model -> model.setBodyOfMethod("statement"))
+	//tells builder that all variables defined out of scope of template model (body of the method)
+	//are considered as template parameters
+	.configureAutomaticParameters()
+	//builds an instance of Pattern
+	.build();
+```
+
+
+This template specifies a
+statements (all statmenets of body of method `statement`) that is a precondition to check that a list 
```

Comment:
typo on statements

Reply:
Thanks, should be fixed with next commit.